packages/fxa-content-server/app/scripts/views/authorization.js

```diff
@@ -19,6 +32,57 @@ class AuthorizationView extends BaseView {
       this.replaceCurrentPage('/oauth/');
     }
   }
+
+  _doPromptNone() {
+    const account = this.getSignedInAccount();
+
+    return (
+      this.relier
+        .validatePromptNoneRequest(account)
+        .then(() => this.useLoggedInAccount(account))
+        // Both validatePromptNoneRequest and useLoggedInAccount can
+        // reject with INVALID_TOKEN, .catch handles both cases.
+        .catch(err => {
+          const normalizedErr = this._normalizePromptNoneError(err);
+          return this._handlePromptNoneError(normalizedErr);
+        })
+    );
+  }
+
+  _normalizePromptNoneError(err) {
+    if (AuthErrors.is(err, 'INVALID_TOKEN')) {
+      err = OAuthErrors.toError('PROMPT_NONE_NOT_SIGNED_IN');
+    }
+
+    return err;
+  }
+
+  _handlePromptNoneError(err) {
+    return Promise.resolve().then(() => {
+      if (this._shouldSendErrorToRP(err)) {
+        // Unless the RP overrides this behavior, errors with a `response_error_code`
+        // redirect back to the RP with `response_error_code` as the `error` parameter.
```

Comment:
Under what circumstances would the RP want to do this?

Reply:
Ah, I see, it's for the functional tests.

Reply:
Added a note to that effect.